core/main/src/broker/http_broker.rs

```diff
@@ -15,78 +15,143 @@
+    }
+}
+async fn send_broker_response(callback: &BrokerCallback, request: &BrokerRequest, body: &[u8]) {
+    match BrokerOutputForwarder::handle_non_jsonrpc_response(
+        body,
+        callback.clone(),
+        request.clone(),
+    ) {
+        Ok(_) => {}
+        Err(e) => {
+            error!("Error message from http broker {:?}", e)
+        }
+    }
+}
+async fn body_to_bytes(body: Body) -> Vec<u8> {
+    match hyper::body::to_bytes(body).await {
+        Ok(bytes) => {
+            let value: Vec<u8> = bytes.into();
+            value.as_slice().to_vec()
+        }
+        Err(e) => format!("error in http broker transforming body to bytes {}", e)
+            .to_string()
+            .as_bytes()
+            .to_vec(),
+    }
+}
 
 impl EndpointBroker for HttpBroker {
     fn get_broker(request: BrokerConnectRequest, callback: BrokerCallback) -> Self {
         let endpoint = request.endpoint.clone();
         let (tx, mut tr) = mpsc::channel(10);
         let broker = BrokerSender { sender: tx };
-        let is_json_rpc = endpoint.jsonrpc;
         let uri: Uri = endpoint.get_url().parse().unwrap();
-        // let mut headers = HeaderMap::new();
-        // headers.insert("Content-Type", "application/json".parse().unwrap());
-        // if let Some(auth) = &endpoint.authentication {
-        //     if auth.to_lowercase().contains("bearer") {
-        //         if let Some(session) = session {
-        //             headers.insert(
-        //                 "Authorization",
-        //                 format!("Bearer {}", session.token).parse().unwrap(),
-        //             );
-        //         }
-        //     }
-        // }
-
         let client = Client::new();
         tokio::spawn(async move {
             while let Some(request) = tr.recv().await {
-                if let Ok(broker_request) = Self::update_request(&request) {
-                    let body = Body::from(broker_request);
-                    let http_request = Request::new(body);
-                    let (mut parts, body) = http_request.into_parts();
-                    parts.method = Method::POST;
-                    parts.uri = uri.clone();
-                    //parts.headers = headers.clone();
-
-                    let http_request = Request::from_parts(parts, body);
-                    if let Ok(v) = client.request(http_request).await {
-                        let (parts, body) = v.into_parts();
+                debug!("http broker received request={:?}", request);
+                match send_http_request(&client, Method::GET, &uri, &request.clone().rule.alias)
+                    .await
+                {
+                    Ok(response) => {
+                        let (parts, body) = response.into_parts();
+
+                        let body = body_to_bytes(body).await;
+                        let mut request = request;
+
+                        let v = vec![serde_json::from_slice::<serde_json::Value>(&body).unwrap()];
```

Comment:
Should we be concerned about unwrapping response data from external sources?

Reply:
We definitely should, and now are :) - Thanks!